mlir/include/mlir/Tools/mlir-opt/MlirOptMain.h

```diff
@@ -136,13 +137,24 @@ class MlirOptMainConfig {
   }
   bool shouldShowDialects() const { return showDialectsFlag; }
 
-  /// Set whether to split the input file based on the `// -----` marker into
-  /// pieces and process each chunk independently.
-  MlirOptMainConfig &splitInputFile(bool split = true) {
-    splitInputFileFlag = split;
+  /// Set the marker on which to split the input into chunks and process each
+  /// chunk independently. Input is not split if empty.
+  MlirOptMainConfig &
+  splitInputFile(std::string splitMarker = kDefaultSplitMarker) {
+    splitInputFileFlag = std::move(splitMarker);
+    return *this;
+  }
+  bool shouldSplitInputFile() const { return splitInputFileFlag.empty(); }
```

Comment:
I would expect that this should return true if splitInputFileFlag is not empty? Otherwise we will try to use empty string as split marker?
I think that IREE makes use of this function, otherwise it seems it is currently untested.

Reply:
Yeah, I think you are right. It should be `!splitInputFileFlag.empty()`.